pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/visualization/treestructure/HistoricalDataTable.kt

```diff
@@ -0,0 +1,149 @@
+// SPDX-FileCopyrightText: 2023 2023
+//
+// SPDX-License-Identifier: MIT
+
+package com.amos.pitmutationmate.pitmutationmate.visualization.treestructure
+
+import com.amos.pitmutationmate.pitmutationmate.reporting.XMLParser
+import com.amos.pitmutationmate.pitmutationmate.services.ReportPathGeneratorService
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.ui.treeStructure.treetable.TreeTable
+import java.awt.GridLayout
+import java.io.File
+import java.nio.file.Path
+import javax.swing.JPanel
+import javax.swing.JScrollPane
+import kotlin.io.path.exists
+
+class HistoricalDataTable(project: Project) : JPanel() {
+
+    private var rootNode: DataNode = DataNode(
+        "",
+        "",
+        "",
+        "",
+        "",
+        mutableListOf()
+    )
+    private var alreadyFoundData: MutableList<String> = mutableListOf()
+    private var treeTable: TreeTable
+    companion object {
+        const val ID = "HistoricalData"
+        const val TITLE = "Historical Data"
+    }
+
+    init {
+        layout = GridLayout(0, 1)
+        val treeTableModel = TreeTableModel(createDataStructure(project))
+        treeTable = TreeTable(treeTableModel)
+        treeTable.setRootVisible(true)
+        treeTable.apply {
+            tree.apply {
+                showsRootHandles = true
+                isRootVisible = true
+            }
+        }
+        treeTable.tableHeader.reorderingAllowed = false
+        treeTable.tableHeader.resizingAllowed = true
+        this.add(JScrollPane(treeTable))
+        setSize(1000, 800)
+    }
+
+    private fun gatherHistoricData(
+        coverageReports: MutableList<XMLParser.CoverageReport>,
+        packageReports: MutableList<XMLParser.CoverageReport>,
+        totalReports: MutableList<XMLParser.CoverageReport>,
+        directory: File
+    ) {
+        val maxNumberOfLastReports = 3
+
+        if (directory.exists() && directory.isDirectory) {
+            val parser = XMLParser()
```

Comment:
Could you move the xml parser things into the mutation result service?
As in *getHistoricalMutationResult* or so?

This would increase the readability of this class and serve as a separation of concerns